src/metabase/models/database.clj

```diff
@@ -59,14 +59,24 @@
     ;; schedule the Database sync & analyze tasks
     (schedule-tasks! database)))
 
-(defn- post-select [{driver :engine, :as database}]
-  (cond-> database
-    ;; TODO - this is only really needed for API responses. This should be a `hydrate` thing instead!
-    (driver.impl/registered? driver)
-    (assoc :features (driver.u/features driver database))
+(def ^:private ^:dynamic *normalizing-details*
+  "Track whether we're calling [[driver/normalize-db-details]] already to prevent infinite
+  recursion. [[driver/normalize-db-details]] is actually done for side effects!"
+  false)
```

Comment:
I don't know why Toucan 1 didn't trigger this infinite loop, but my best guess is that it was a bug. Worked around it here